### PR TITLE
🌱 Add golangci-lint file

### PR DESCRIPTION
**Pull request by kashifest**

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+          alias: clusterv1alpha4
+        - pkg: sigs.k8s.io/cluster-api/api/v1beta1
+          alias: clusterv1
+        # CAPM3
+        - pkg: github.com/metal3-io/cluster-api-provider-metal3/api/v1alpha5
+          alias: infrav1alpha5
+        - pkg: github.com/metal3-io/cluster-api-provider-metal3/api/v1beta1
+          alias: infrav1
+        # BMO
+        - pkg: github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1
```
Comment (furkatgofurov7) [inline .golangci.yaml:90]: have you verified if all these packages are used in the code base?

File: .golangci.yaml [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    - unnecessaryDefer
+    - whyNoLint
+    - wrapperFunc
+  unused:
+    go: "1.17"
+issues:
+  exclude-rules:
+    - path: test/e2e
+      linters:
+        - gosec
```
Comment (lentzi90) [inline .golangci.yaml:123]: This is probably not needed.

**Review by lentzi90 — APPROVED** [review]

File: .golangci.yaml (hunk 1 above)
Comment (kashifest) [reply .golangci.yaml:90]: ow now, I have just copy-pasted capm3 one, so I need to scan through go.mod file to check it these exists right ?

Comment (furkatgofurov7) [reply .golangci.yaml:90]: not go mod better look through of files if all of those packages are imported or not.

Comment (kashifest) [reply .golangci.yaml]: Done

File: .golangci.yaml (hunk 2 above)
Comment (kashifest) [reply .golangci.yaml]: Done

File: docs/deployment_workflow.md [hunk 3]
```diff
@@ -4,14 +4,14 @@
...
-* CAPI
+* clusterv1
 * IPAM
 
 The IPAM controller has a dependency on Cluster API *Cluster* objects.
 
 ## Requirements
 
-CAPI CRDs and controllers must be deployed and the cluster objects exist for
+clusterv1 CRDs and controllers must be deployed and the cluster objects exist for
```
Comment (lentzi90) [inline docs/deployment_workflow.md:14]: Too much find/replace I think! :smile:

Comment (kashifest) [reply docs/deployment_workflow.md]: Indeed, thanks, done

**Review by lentzi90 — APPROVED** [review]

**Comment by kashifest** [comment]
I think the addition of golangci is somehow making the tests heavier and this timeouts are happening, lets merge this and if it happens again , I will increase the timeout . 
/retest

**Review by furkatgofurov7 — APPROVED** [review]

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Review by Rozzii — APPROVED** [review]

**Pull request merged**